### ISSUE TITLE
Doe tensorflow has api  similar to pytorch's “masked_fill_”

### ISSUE THREAD

**Issue by jason-liew** [tensorflow/tensorflow#41617]
pytorch has squeeze , tf also has squeeze.
pytorch has unsqueeze, tf has expand_dims do the same thing with diff name.
pythorh has masked_fill , tf has ?

<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using):  2.2.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
   [masked_fill_ api doc in pytorch](https://pytorch.org/docs/stable/tensors.html?highlight=masked_fill_#torch.Tensor.masked_fill_)
**Will this change the current api? How?**
   Maybe not. It will add a new api. 
**Who will benefit with this feature?**
    someone try to migrate from pytorch to tensorflow like me.
**Any Other info.**

**Comment by ravikyram** [comment]
@jason-liew 

Can you please go through [tf.boolean_mask](https://www.tensorflow.org/api_docs/python/tf/boolean_mask) and see whether you are looking for similar functionality.Thanks!

**Comment by jason-liew** [comment]
No,it's different.

tensorflow:
```
 t1 = tf.constant([1,2,3])
 t1_mask = numpy.array([1,0,1])
 tf.boolean_mask(t1,t1_mask) # [1, 3]
```
pytorch:
```
p1 = torch.tensor([1,2,3])
p1_mask = torch.tensor([1,0,1])
p1.masked_fill_(p1_mask,5) # [5, 2, 5]
```

**Comment by ravikyram** [comment]
@jason-liew 

Got it.Thanks!
Are you interested to submit a PR?

**Comment by aavishkarmishra** [comment]
@gowthamkpr Hi, I would like to work in this can you help with how and from where I can start ?

**Comment by aavishkarmishra** [comment]
@gowthamkp Should we have to add a new api or we can also override boolean_mask with one more parameter which one will be better?

**Comment by rohan100jain** [comment]
Strawman proposal. Can this be implemented as

def masked_fill(t, mask, value):
  return t * (1 - tf.cast(mask, tf.float32)) + value * tf.cast(mask, tf.float32)

Might have to do some dtype handling but roughly should work? What are the performance requirements for this?

**Comment by aavishkarmishra** [comment]
@ravikyram @gowthamkpr please review this PR #41975

**Comment by fsx950223** [comment]
```python
tf.where([True, False, True],5,[1,2,3])
```

**Comment by alextp** [comment]
We recommend the tf.where implementation.
